ocaml/testsuite/tests/typing-modes/modes.ml

```diff
@@ -376,3 +376,179 @@ Line 2, characters 38-41:
                                           ^^^
 Error: The extension "mode" is disabled and cannot be used
 |}]
+
+
+(*
+ * Modification of return modes in argument position
+ *)
+
+let use_local (f : _ -> _ -> _ @@ local) x y =
+  f x y
+let result = use_local (^) "hello" " world"
+[%%expect{|
+val use_local : local_ ('a -> 'b -> 'c) -> 'a -> 'b -> 'c = <fun>
+val result : string = "hello world"
+|}]
+
+let use_local_ret (f : _ -> _ @ local) x y =
```

Comment:
Is this actually OK to accept, given the extra meaning of `local` in return position? That is, `use_local_ret` will assume `f` allocates in the parent stack frame, but it won't. Our calling convention is OK with this?

Reply:
It’s fine, we only need to avoid unexpectedly allocating in a caller. This isn’t new behaviour in this PR though, or even a new test, I just moved the test from elsewhere.